samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -133,22 +136,73 @@ public ApplicationStatus status(StreamApplication app) {
 
       if (hasNewJobs) {
         // There are jobs not started, report as New
-        return ApplicationStatus.New;
+        return New;
       } else if (hasRunningJobs) {
         // All jobs are started, some are running
-        return ApplicationStatus.Running;
+        return Running;
       } else if (unsuccessfulFinishStatus != null) {
         // All jobs are finished, some are not successful
         return unsuccessfulFinishStatus;
       } else {
         // All jobs are finished successfully
-        return ApplicationStatus.SuccessfulFinish;
+        return SuccessfulFinish;
       }
     } catch (Throwable t) {
       throw new SamzaException("Failed to get status for application", t);
     }
   }
 
+  /* package private */ ApplicationStatus getApplicationStatus(JobConfig jobConfig) {
+    JobRunner runner = new JobRunner(jobConfig);
+    ApplicationStatus status = runner.status();
+    LOG.debug("Status is {} for job {}", new Object[]{status, jobConfig.getName()});
+    return status;
+  }
+
+  /**
+   * Waits until the application finishes.
+   */
+  public void waitForFinish() {
+    waitForFinish(Duration.ofMillis(0));
+  }
+
+  /**
+   * Waits until the application finishes. It times out after the input duration has elapsed.
+   * Provide a value less than 1 for input duration to wait indefinitely.
+   *
+   * @param timeout time to wait for the application to finish
```

Comment:
let's not throw here. Instead return the AppStatus here.

Reply:
discussed offline. We are going to return a boolean to denote if the method timed out or not.